### DISCUSSION TITLE
Is it possible to include types for the object structure returned by the REST API in the generate:types file?

### DISCUSSION THREAD

**Discussion by Bash4195** [payloadcms/payload#2768] [Q&A]
I have a NextJS frontend making an API GET request to Payload which is returning some data. I'm using the `generate:types` command to output my collection types into a file in my NextJS folder so I can use those types on the frontend. I see Payload has this file: https://github.com/payloadcms/payload/blob/7c03e5510cb6bb60c035a5e8e747cb4f1a399dc3/src/mongoose/types.ts

Is it possible to have the generate:types include other things in this file like the type returned from it's REST API?

**Comment by DanRibbens** [comment]
You can import from Payload and then add your own types and pass the generic type from your generated types.
I would rather keep the generation small and composable than generate all the possibilities.
